src/debug_renderer.rs

```diff
@@ -25,6 +25,28 @@ pub struct DebugRenderer<R: gfx::Resources> {
 
 impl<R: gfx::Resources> DebugRenderer<R> {
 
+    pub fn from_canvas<
+        C: gfx::CommandBuffer<R>,
+        F: Factory<R>,
+        O: gfx::render::target::Output<R>,
+        D: Device<Resources = R, CommandBuffer = C>,
+    > (
+        canvas: &mut gfx::Canvas<O, D, F>,
+        frame_size: [u32; 2],
```

Comment:
no need for the size either. `Canvas::output` implements `gfx::Output`, which you can ask for `.get_size()`